source/main.py

```diff
@@ -1092,6 +1092,12 @@ def __init__(self, file_name):
             if prefix == '/': break
             prefix = os.path.dirname(prefix)
 
+        if self.source is None:
```

Comment:
For better error reporting, we usually don't allow a subroutine/class to abort the whole program, except explicitly in main. These exit points could hide deeper and deeper till you lost control when to exit your program.

Pushing error checking into the class as a whole is a good design though. But still, we need to decide to exit(1) the program in main.

To achieve that, we can either return an error code, or throw and exception and catch it in main.